src/kakarot/instructions/system_operations.cairo

```diff
@@ -94,4 +95,44 @@ namespace SystemOperations {
             ctx, new_return_data_len=size.low, new_return_data=new_return_data
         );
     }
+
+    // @notice REVERT operation.
+    // @dev
+    // @custom:since Byzantium
+    // @custom:group System Operations
+    // @custom:gas 0 + dynamic gas
+    // @custom:stack_consumed_elements 0
+    // @custom:stack_produced_elements 0
+    // @return The pointer to the updated execution context.
+    func exec_revert{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(ctx: model.ExecutionContext*) {
+        alloc_locals;
+
+        // Get stack and memory from context
+        let stack = ctx.stack;
+        let memory = ctx.memory;
+
+        // Stack input:
+        // 0 - size: byte size to copy
+        // 1 - offset: byte offset in the memory in bytes
+        let (stack, popped) = Stack.pop_n(self=stack, n=2);
+        // TODO: implement loading of the revert reason based on size value,
+        // currently limited by short string size
+        let size = popped[0];
+        let offset = popped[1];
+
+        // Load revert reason from offset
+        let revert_reason_uint256 = Memory.load(memory, offset.low);
```

Comment:
and maybe here a `load_n` then?

Reply:
what is load_n ?

Reply:
`load_n` like to load `n` bytes from memory. `load` actually loads "32" but this could easily be refactored in `load_n` for any `n`, and using `load_n(..., 32)` for the actual `Memory.exec_load`

Reply:
agreed, as this would be more optimal, it's not part of this PR's concern though. Let's create a different issue for it:)